lib/xapi-stdext-date/date.ml

```diff
@@ -62,20 +71,10 @@ let of_float x =
     time.Unix.tm_sec |> of_string
 
 (* Convert tm in localtime to calendar time, x *)
-let to_float_localtime x =
-  let datetime_to_float y mon d h min s =
-    fst Unix.(mktime { tm_year = y - 1900;
-                       tm_mon = mon - 1;
-                       tm_mday = d;
-                       tm_hour = h;
-                       tm_min = min;
-                       tm_sec = s;
-                       (* These are ignored: *)
-                       tm_wday = 0; tm_yday = 0; tm_isdst = true;
-                     })
-  in
-  let ((y, mon, d), ((h, min, s), _)) = Ptime.to_date_time x in
-  datetime_to_float y mon d h min s
+let to_float_localtime (t, _) = Ptime.to_float_s t
+
+let localtime () =
+  (Ptime_clock.now (), Ptime_clock.current_tz_offset_s ())
```

Comment:
I think `to_float` below can be simplified too, there is no need for a `to_float_localtime` anymore, just use Ptime there directly and ignore the offset (the localtime stuff seems to have been there to make sure you get back UTC, but you already store things in UTC).